core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -3720,7 +3720,7 @@ public RelBuilder sort(Iterable<? extends RexNode> nodes) {
   }
 
   /** Creates a {@link Sort} by expressions, with limit and offset. */
-  public RelBuilder sortLimit(long offset, long fetch, RexNode... nodes) {
+  public RelBuilder sortLimit(Number offset, Number fetch, RexNode... nodes) {
```

Comment:
I think you can leave this API unchanged.
Moreover, it's a public function.
Number includes Double, so it's a bit too much.
It's fine on the getter side.

Reply:
you should probably keep public methods and add new ones only

Reply:
Should I change it back to the int version and add a new Number version as well?